server/athenian/api/align/goals/dates.py

```diff
@@ -37,3 +42,33 @@ def goal_initial_query_interval(
     initial_from = valid_from - current_span
 
     return (initial_from, initial_at)
+
+
+Intervals = tuple[datetime, ...]
+
+
+@dataclasses.dataclass(slots=True, frozen=True)
+class GoalTimeseriesSpec:
+    """The specification for a timeseries collecting goal metric values."""
+
+    intervals: Intervals
+    granularity: GoalSeriesGranularity
+
+    @classmethod
+    def from_timespan(cls, valid_from: datetime, expires_at: datetime) -> GoalTimeseriesSpec:
+        """Build the intervals given the timespan of the goal, as saved on DB."""
+        granularity = cls._granularity(valid_from, expires_at)
+        # Granularity.split accepts an interval where the last day is included
+        from_date, at_date = goal_datetimes_to_dates(valid_from, expires_at)
+
+        date_intervals = tuple(Granularity.split(granularity.value, from_date, at_date))
+        dt_intervals = tuple(
+            datetime.combine(d, time.min, tzinfo=timezone.utc) for d in date_intervals
+        )
+        return GoalTimeseriesSpec(dt_intervals, granularity)
+
+    @classmethod
+    def _granularity(cls, valid_from: datetime, expires_at: datetime) -> GoalSeriesGranularity:
+        if (expires_at - valid_from) <= timedelta(days=93):  # goal with a quarter timespan or less
```

Comment:
Yes, "The granularity is weekly if the goal is set on a quarter, and monthly if the goal is set on a year."
They only have quarter and year goals in UI

Reply:
The calendar month is usually 30 days everywhere (e.g. numpy), hence I am asking about the exact value of 93

Reply:
31 x 3 ? Anyway in a quarter we never have 3 31-days months so we could also set 92.

Reply:
The point is that we are  not defining this strictly since we have only quarter and year goals
When we'll have goals of 4 or 5 months we can decide what's better

Reply:
Interesting. I wasn't exactly right.
```
In [5]: import numpy as np

In [6]: np.timedelta64(3, "M")
Out[6]: numpy.timedelta64(3,'M')

In [7]: np.timedelta64(3, "M").astype("timedelta64[D]")
Out[7]: numpy.timedelta64(91,'D')
```

Reply:
How about 91 to match numpy?

Reply:
Wouldn't this be a problem ?
```
In [2]: (datetime(2001, 10, 1) - datetime(2001, 7, 1)).days
Out[2]: 92
```

Reply:
I don't think so because choosing different dates will give 91. This is a convention...

Reply:
The same way as when they select Q2 we will trigger earlier/later than needed. I just want us to be consistent with the existing code.